build/tools/Automation/DotNet/DotNet.cs

```diff
@@ -0,0 +1,236 @@
+                    "platforms/dotnet/versions/stretch/versionsToBuild.txt",
+            };
+            foreach (string versionsToBuildTxtFile in versionsToBuildTxtFiles)
+            {
+                string line = $"\n{platformConstant.Version}, {platformConstant.Sha},";
+                File.AppendAllText(versionsToBuildTxtFile, line);
+
+                // sort
+                Console.WriteLine($"Updating {versionsToBuildTxtFile}...");
+                var contents = File.ReadAllLines(versionsToBuildTxtFile);
+                Array.Sort(contents);
+                File.WriteAllLines(versionsToBuildTxtFile, contents);
+            }
+        }
+
+        private static Dictionary<string, Constant> GetYamlDotNetConstants(List<Constant> yamlContents)
+        {
+            Dictionary<string, Constant> dotNetConstants = new Dictionary<string, Constant>();
+            foreach (var constant in yamlContents)
+            {
+                if (constant.Name == Constants.DotNetSdkKey ||
+                    constant.Name == Constants.DotNetRuntimeKey)
+                {
+                    dotNetConstants.Add(constant.Name, constant);
+                }
+            }
+            return dotNetConstants;
+        }
+
+        private static string GenerateDotNetConstantKey(PlatformConstant platformConstant)
+        {
+            string[] splitVersion = platformConstant.Version.Split('.');
+            string majorVersion = splitVersion[0];
+            string minorVersion = splitVersion[1];
+            string majorMinor = majorVersion + minorVersion;
+            string constant;
+            if (platformConstant.VersionType.Equals(Constants.SdkName))
+            {
+                // TODO: add try catch in case the integer is un-parseable.
+                int majorVersionInt = int.Parse(majorVersion);
+
+                // dotnet/dotnetcore are used based on the major version
+                string prefix = majorVersionInt < 5 ? $"dot-net-core" : "dot-net";
+                constant = $"{prefix}-{majorMinor}-sdk-version";
+            }
+            else
+            {
+                constant = $"{platformConstant.VersionType}-app-{majorMinor}";
+            }
+
+            // TODO: add Logger.Debug the constant that is generated
+            return constant;
+        }
+
+        private static string GetSha(List<FileObj> files)
+        {
+            HashSet<string> tarFileNames = new HashSet<string>() {
+                "dotnet-sdk-linux-x64.tar.gz",
+                "dotnet-runtime-linux-x64.tar.gz",
+                "aspnetcore-runtime-linux-x64.tar.gz",
+            };
+            foreach (var file in files)
+            {
+                if (tarFileNames.Contains(file.Name))
+                {
+                    return file.Hash;
+                }
+            }
+
+            // TODO: special exception if sha not found
+            Console.WriteLine("No sha found");
+
+            return string.Empty;
```

Comment:
We should throw an exception now and handle it by the caller so that they can choose to skip processing the current version if it doesn't contain a 64-bit Linux release that we could use since right now we're still going ahead and creating a `PlatformConstant` object with an empty string `Sha` property.

Reply:
Updated